### ISSUE TITLE
Assignment Error: RHS TypedDict is a subset of the LHS TypedDict

### ISSUE THREAD

**Issue by grant-zietsman** [microsoft/pyright#1105]
**Describe the bug**
Unable to assign TypedDicts with the same signatures except that the LHS TypedDict has a field with a Union or Option type and the RHS TypedDict has the same field but with the Union or Option type refined.

**To Reproduce**
See minimal example below.

**Expected behavior**
I would have expected the assignment to be valid.

**Screenshots or Code**
```py
from typing import TypedDict, Union

class X(TypedDict):
    a: Union[int, str]

class Y(TypedDict):
    a: int

y: Y = {"a": 1}
x: X = y
#  ^^^^^
# Expression of type "Y" cannot be assigned to declared type "X"
#   "a" is an incompatible type Pylance (reportGeneralTypeIssues)
```
![image](https://user-images.githubusercontent.com/7155647/96336682-c2cd6b80-1081-11eb-9a02-7c1af478ad84.png)

**VS Code extension or command-line**
```sh
> code --list-extensions --show-versions | grep pylance

ms-python.vscode-pylance v2020.10.1

> code --version

1.50.1
d2e414d9e4239a252d1ab117bd7067f125afd80a
x64
```

**Additional context**

Python 3.8.5

**Comment by erictraut** [comment]
This is the correct behavior according to [PEP 589](https://www.python.org/dev/peps/pep-0589/), which defines how `TypedDict` works. The class `Y` and `X` are distinct classes, and `Y` does not inherit from `X`, so you can't assign a variable of type `Y` to a variable of type `X`. I verified that other type checkers report the same error in this case.
